url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/agentes/jade/Centralizador/AgenteCentralizador.java

```diff
@@ -0,0 +1,415 @@
+import gramatica.Centralizador.Absyn.ECollect2;
+import gramatica.Centralizador.Absyn.EData1;
+import gramatica.Centralizador.Absyn.EData2;
+import gramatica.Centralizador.Absyn.EData3;
+import gramatica.Centralizador.Absyn.EData4;
+import gramatica.Centralizador.Absyn.ERemedy1;
+import gramatica.Centralizador.Absyn.ERemedy2;
+import gramatica.Monitor.Absyn.EDados;
+import gramatica.Monitor.Absyn.EDados1;
+import gramatica.Monitor.Absyn.EDados2;
+import gramatica.Monitor.Absyn.EDados3;
+import jade.core.AID;
+import jade.core.Agent;
+import jade.core.behaviours.CyclicBehaviour;
+import jade.core.behaviours.TickerBehaviour;
+import jade.lang.acl.ACLMessage;
+import jade.lang.acl.MessageTemplate;
+import jade.wrapper.AgentContainer;
+import jade.wrapper.StaleProxyException;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.List;
+
+import util.GeradorAleatorioMsg;
+import util.UpdateAgentList;
+import agentes.jade.Monitor.Afericao;
+import agentes.jade.Monitor.GrammarParserMonitor;
+import agentes.jade.Monitor.TarefaMonitor;
+
+public class AgenteCentralizador extends Agent {
+	private static final int INICIO_FEBRE = 38;
+	private static final int MIN_PRESSAO_DIAS_BOA = 60;
+	private static final int MIN_PRESSAO_SIST_BOA = 100;
+	private static final int MAX_PRESSAO_DIAS_BOA = 90;
+	private static final int MAX_PRESSAO_SIST_BOA = 140;
+	private final static int MIN_HEMO_BOA = 14;
+	private final static int MAX_HEMO_BOA = 18;
+	private final static int MIN_BILI_BOA = 0;
+	private final static int MAX_BILI_BOA = 5;
+	
+	private static final int INTERVALO_REQUISICAO = 8000;
+	private final int AGENTSNUMBER = 2;
+	
+	private AtuadorBusyControl atuadorIsBusy[] = new AtuadorBusyControl[AGENTSNUMBER];
+	
+	private List<AID> monitor;
+	private List<AID> atuador;
+	
+	private String monitorName, atuadorName;
+	
+	private File file;
+	
+	/*
+	 * TODO LIST
+	 * TODO Implementar metodo que transforma objeto em mensagem de texto string
+	 * TODO Implementar geraÃ§Ã£o automatica de mensagens randomicas a partir da gramatica
+	 * TODO Implementar logica de decisao do centralizador
+	 * TODO Implementar armazenamento e tabulacao dos dados no centralizador
+	 * TODO Centralizador precisa saber de tudo que estÃ¡ fazendo, exemplos:
+	 * 1 - Estou aplicando remedios agora? quais? NÃ£o vou pedir para aplciar a mesma coisa se jÃ¡ estou aplicando
+	 * 2 - Estou medindo alguma cosia agora? o que? Nao vou pedir para medir a mesma coisa se jÃ¡ estou medindo
+	 * 
+	 * */
+	
+	public void setup() {
+		System.out.println(getLocalName() +  ": Sou o agente " + getAID().getLocalName() + " e estou pronto.");
+
+		// create agent t1 on the same container of the creator agent
+		AgentContainer container = (AgentContainer) getContainerController();
+   
+		// create the file on project directory
+		file =new File("outputData.txt");
+		try {
+			file.createNewFile();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

`File.createNewFile` returns `false` if the named file already exists and an empty file is not created. Check the value returned by `createNewFile()`. Handle when the method returns `false`. Otherwise, you might silently overwrite an existing file which could lead to data loss. See the following diffs on GitHub that make this code change: [diff1](https://github.com/lingochamp/FileDownloader/commit/65069c18a4b1b4cfa758173ef49d69c6897fc77a#diff-2b6ad96df9bf9828c22f961e9e9a5b933ab2df7b2fb0f12d62c044309e1e2842L331), [diff2](https://github.com/m-szalik/javamail/commit/62b1cb45e32a998efbe6cfa09b06752869f590ae#diff-440ef5498fc2c802274cf136e7bd7d682bbc10c5976afb5af718984e24deced4L45)

Learn more: [File.createNewFile in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#createNewFile--)